### PR TITLE
CI to detect out of sync migrations

### PR DESCRIPTION
**Pull request by chrisclark**
Making sure #633 doesn't happen again.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,6 +53,9 @@ jobs:
           name: coverage-data-${{ matrix.python-version }}
           path: '${{ github.workspace }}/.coverage*'
 
+      - name: Check migrations
+        run: python manage.py makemigrations --check
```
Comment (marksweb) [inline .github/workflows/test.yml:57]: You can add this to the test suite as well.

https://adamj.eu/tech/2024/06/23/django-test-pending-migrations/

Comment (marksweb) [reply .github/workflows/test.yml:57]: Actually it's already part of the pre-commit hooks so in theory you shouldn't be able to commit code that needs migrations and doesn't include them.

Comment (jefer94) [reply .github/workflows/test.yml:57]: Pre-commit should be bypassed with a param on the console if someone didn't install Pre-commit

Comment (chrisclark) [reply .github/workflows/test.yml]: I'd like to address this particular issue outside of precommit. Interesting idea to do it in tests directly...I might go that direction seeing as it's a bit janky to put it in the pypi-test as I've done here (just did it to save time during CI to piggyback on the environment)

**Comment by chrisclark** [comment]
This turned into a pretty interesting PR. It turns out that tox was installing all of the dev dependencies for all of the various configurations -- which is why (for example) I had an import issue in the 5.0 release (a dev dependency turned out to be required, even for a base install). I've reconfigured the tox setup to properly test with JUST the base deps, and then also with 'everything'. This gets better coverage on the 'everything' version (hits more test cases) and runs faster (because it's actually installing only the base deps for half of the test runs). Cool! Also required skipping certain tests for the base tests -- and I actually caught another import problem with the base install!

File: .github/workflows/test.yml (hunk 1 above)
Comment (marksweb) [reply .github/workflows/test.yml]: > Pre-commit should be bypassed with a param on the console, and if someone didn't install Pre-commit

Yes we need pre-commit either running manually as a workflow run, or integrating the bot from pre-commit.ci

I think we've already got PRs from people who don't have pre-commit installed.

**Pull request merged**